connectivity/common.h

```diff
@@ -0,0 +1,11 @@
+#ifndef CONN_COMMON_H
+#define CONN_COMMON_H
+#include "common/ta_errors.h"
+
+status_t ta_url_matcher(char const *const url, char *const regex_rule);
```

Comment:
how about rename it as `api_path_matcher()`?

Reply:
`ta_url_matcher()` does things more than match url, because the api path of mqtt is not actually URL